src/test/java/com/fittracker/fittracker/controller/ProductControllerIntegrationTest.java

```diff
@@ -88,6 +95,63 @@ void givenValidGetRequestWithNonexistentId_shouldReturnError() throws Exception
             assertThat(response).isEqualTo(expectedResponse);
         }
 
+
+    }
+
+    @Nested
+    class Put {
+
+        @BeforeEach
+        void beforeEach() {
+            productRepository.save(product());
```

Comment:
It would allow to remove `@BeforeEach` portion form `PUT` part of tests, but additional element in a database would make us modify `POST` and `GET` tests (assertion about elements, getting id, getting latest added product). I think that at least for now `@BeforeEach` lets us keep it simple and not confusing.

On the other hand we can think if integration tests with emptied database at the start reflect "integrational" character of these tests. I think It's worth to have a chat about that.